src/handlers/swap_handlers.ts

```diff
@@ -452,6 +458,22 @@ const parseSwapQuoteRequestParams = (req: express.Request, endpoint: 'price' | '
         ]);
     }
 
+    let priceImpactProtectionPercentage = DEFAULT_PRICE_IMPACT_PROTECTION_PERCENTAGE;
+    if (req.query.priceImpactProtectionPercentage !== undefined) {
+        PRICE_IMPACT_PROTECTION_SPECIFIED.inc();
+        priceImpactProtectionPercentage = Number.parseFloat(req.query.priceImpactProtectionPercentage as string);
```

Comment:
I think we need to add a check for `NaN`.

Reply:
we don't seem to check for NaN in slippagePercentage either...

Reply:
whether the return value of ` Number.parseFloat` is `NaN`

Reply:
actually looks like the same regex that catches negative numbers also checks to make sure its some kind of number. i attempted to test the requested changes and it was caught earlier by the regex

Reply:
oh i see -- makes sense

It would have been great if we have a single validator (vs two step process -- error description isn't very great when it fails regex-based validation)